src-java/kilda-model/src/main/java/org/openkilda/model/HaFlow.java

```diff
@@ -0,0 +1,593 @@
+            return null;
+        }
+        return getPath(getForwardPathId()).orElse(null);
+    }
+
+    /**
+     * Add a path and set it as the forward path.
+     */
+    public void setForwardPath(HaFlowPath forwardPath) {
+        if (forwardPath != null) {
+            validateForwardPath(forwardPath);
+            if (!hasPath(forwardPath)) {
+                addPaths(forwardPath);
+            }
+            setForwardPathId(forwardPath.getHaPathId());
+        } else {
+            setForwardPathId(null);
+        }
+    }
+
+    /**
+     * Get the protected forward path.
+     */
+    public final HaFlowPath getProtectedForwardPath() {
+        if (getProtectedForwardPathId() == null) {
+            return null;
+        }
+        return getPath(getProtectedForwardPathId()).orElse(null);
+    }
+
+    /**
+     * Add a path and set it as the protected forward path.
+     */
+    public void setProtectedForwardPath(HaFlowPath forwardPath) {
+        if (forwardPath != null) {
+            validateForwardPath(forwardPath);
+            if (!hasPath(forwardPath)) {
+                addPaths(forwardPath);
+            }
+            setProtectedForwardPathId(forwardPath.getHaPathId());
+        } else {
+            setProtectedForwardPathId(null);
+        }
+    }
+
+    /**
+     * Get the reverse path.
+     */
+    public HaFlowPath getReversePath() {
+        if (getReversePathId() == null) {
+            return null;
+        }
+        return getPath(getReversePathId()).orElse(null);
+    }
+
+
+    /**
+     * Add a path and set it as the reverse path.
+     */
+    public void setReversePath(HaFlowPath reversePath) {
+        if (reversePath != null) {
+            if (!hasPath(reversePath)) {
+                validateReversePath(reversePath);
+                addPaths(reversePath);
+            }
+            setReversePathId(reversePath.getHaPathId());
+        } else {
+            setReversePathId(null);
+        }
+    }
+
+    /**
+     * Get the protected reverse path.
+     */
+    public HaFlowPath getProtectedReversePath() {
+        if (getProtectedReversePathId() == null) {
+            return null;
+        }
+        return getPath(getProtectedReversePathId()).orElse(null);
+    }
+
+    /**
+     * Add a path and set it as the protected reverse.
+     */
+    public void setProtectedReversePath(HaFlowPath reversePath) {
+        if (reversePath != null) {
+            if (!hasPath(reversePath)) {
+                validateReversePath(reversePath);
+                addPaths(reversePath);
+            }
+            setProtectedReversePathId(reversePath.getHaPathId());
+        } else {
+            setProtectedReversePathId(null);
+        }
+    }
+
+    private void validateEndpoints(HaFlowPath path) {
+        if (!path.getSharedSwitchId().equals(getSharedEndpoint().switchId)) {
+            throw new IllegalArgumentException(format("HA-path %s has shared endpoint switch id %s, but %s expected",
```

Comment:
ID is short for `identification`; it is an abbreviation, usually spelled upper case. 
An article is missing: "... has **_the_** shared ...". Passive voice in the second part: " ... but %s **_is_** expected"